pyrk/th_component.py

```diff
@@ -83,33 +131,213 @@ def update_temp(self, timestep, temp):
         :param timestep: the timestep at which to query the temperature
         :type timestep: int
         :param temp: the new tempterature
-        :type float: float, units of kelvin
+        :type float: float, dimensionless
         """
         self.T[timestep] = temp
         self.prev_t_idx = timestep
         return self.T[timestep]
 
-    def dtemp(self):
-        if self.prev_t_idx == 0:
-            return 0.0*units.kelvin
-        else:
-            return (self.T[self.prev_t_idx] - self.T[self.prev_t_idx-1])
+    def dtemp(self, timestep, T0_timestep):
```

Comment:
docstring for timestep and T0 timestep needed  (that is, I think you're not asking for the time. Rather, the index of the time. Elsewhere in the code (in particular where temp_reactivity is actually used) this variable has been called t_idx and t_feedback_idx . I think that helps distinguish that the number we need is the index.